tests/transformation/test_nn_resize_to_deconv.py

```diff
@@ -0,0 +1,173 @@
+    new_model = model.transform(ResizeConvolutionToDeconvolution(maintain_bit_width=maintain_bit_width))
+    # check that there are no Resize ops left
+    op_types = list(map(lambda x: x.op_type, new_model.graph.node))
+    assert "Resize" not in op_types, "Error: the Resize nodes should be removed."
+    # get the output from the deconvolution model
+    output_deconv = oxe.execute_onnx(new_model, input_dict)[oname]
+    # maintaining the specified bit width introduces additional clipping errors that
+    # shouldn't be expected to maintain reasonable functional similarity
+    if not maintain_bit_width:
+        assert np.isclose(
+            output_deconv, output_resize_conv, atol=1 / 255.0, rtol=1.0
+        ).all(), "Error: expected output does not match the produced output."
+
+
+def create_nn_resize_conv_model(
+    in_channels: int, out_channels: int, input_dim: int, kernel_size: int, upscale_factor: int, bias: bool
+):
+    assert isinstance(kernel_size, int), "Assuming square kernels, so kernel_size needs to be an int."
+    padding = (kernel_size - 1) // 2
+
+    ifm_ch = in_channels
+    ifm_dim = input_dim
+    ofm_dim = ifm_dim * upscale_factor
+    ofm_ch = out_channels
+    scales = np.array([1.0, 1.0, upscale_factor, upscale_factor], dtype=np.float32)
+
+    resize = oh.make_node(
+        "Resize",
+        inputs=["inp", "roi", "scales"],
+        outputs=["hid"],
+        mode="nearest",
+    )
+    conv = oh.make_node(
+        op_type="Conv",
+        inputs=["hid", "W"] if not bias else ["hid", "W", "B"],
+        outputs=["out"],
+        kernel_shape=[kernel_size, kernel_size],
+        pads=[padding, padding, padding, padding],
+        strides=[1, 1],
+        group=1,
+        dilations=[1, 1],
+    )
+
+    input_shape = [1, ifm_ch, ifm_dim, ifm_dim]
+    output_shape = [1, ofm_ch, ofm_dim, ofm_dim]
+
+    conv_param_shape = [ofm_ch, ifm_ch, kernel_size, kernel_size]
+    bias_param_shape = [ofm_ch]
+
+    inp = oh.make_tensor_value_info("inp", TensorProto.FLOAT, input_shape)
+    out = oh.make_tensor_value_info("out", TensorProto.FLOAT, output_shape)
+
+    W_conv = oh.make_tensor_value_info("W", TensorProto.FLOAT, conv_param_shape)
+    B_conv = oh.make_tensor_value_info("B", TensorProto.FLOAT, bias_param_shape)
+
+    value_info = [W_conv] if not bias else [W_conv, B_conv]
+
+    graph = oh.make_graph(
+        nodes=[resize, conv],
+        name="cnv_graph",
+        inputs=[inp],
+        outputs=[out],
+        value_info=value_info,
+    )
+    modelproto = qonnx_make_model(graph, producer_name="test_model")
+    model = ModelWrapper(modelproto)
+    model.set_initializer("roi", np.empty(0))
+    model.set_initializer("scales", scales)
+    model.set_initializer("W", np.random.rand(*conv_param_shape).astype(np.float32))
+    if bias:
+        model.set_initializer("B", np.random.rand(*bias_param_shape).astype(np.float32))
+    model.save("cnv-graph.onnx")
```

Comment:
please remove generated files after test is done, unless you want to leave this intact for debugging in case the test fails